src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -296,6 +302,11 @@ private bool ShouldIncludeStateSet(Project project, StateSet stateSet)
                     return false;
                 }
 
+                if (_shouldIncludeAnalyzer != null && !_shouldIncludeAnalyzer(stateSet.Analyzer))
+                {
+                    return false;
+                }
```

Comment:
This is the core fix. Most of the other changes in the PR are just threading through this func.